internal/provider/resource_apply_module.go

```diff
@@ -406,21 +477,48 @@ func (t *applyModuleResource) doRun(ctx context.Context,
 	// while plannedRun.Plan.Status is "finished".
 	//
 	if !strings.HasPrefix(string(plannedRun.Status), "planned") {
+
+		// Bring in any error message(s) from the finished inner plan run.
+		innerPlanRunDiags := t.extractRunError(ctx, plannedRun)
+		diags.Append(innerPlanRunDiags...)
+		if innerPlanRunDiags.HasError() {
+			return diags
+		}
+
 		diags.AddError("Plan failed", string(plannedRun.Status))
```

Comment:
This message can be "Plan failed with unknown error"

Reply:
Done.